lib/apigateway.js

```diff
@@ -12,9 +12,19 @@ function sleep(s) {
 }
 
 function APIGateway(options) {
-	// Load the credentials
-	AWS.config.region = options.region || 'us-west-1';
-	AWS.config.credentials = new AWS.SharedIniFileCredentials({profile: options.profile});
+	AWS.config.region = options.awsRegion || 'us-west-1';
+
+	// We don't want to use the whole options object in SharedIniFileCredentials so we build up
+	// credentialsOptions
+	let credentialsOptions = {
+		profile: options.awsProfile
+	};
+
+	if (options.awsFilename) {
+		credentialsOptions.filename = options.awsFilename;
+	}
+	// Load the config credentials
+	AWS.config.credentials = new AWS.SharedIniFileCredentials(credentialsOptions);
```

Comment:
Actually today I realized that the credentials are optional. It's perfectly valid to run this module from an ec2 instance with an instance role attached to it. So all of this (except the `region`) is optional. I implemented that in another library I created today and might come in handy. You can almost literaly copy-paste that part.

https://github.com/SamVerschueren/lambda-update-alias/blob/master/index.js#L18-L32